WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt

```diff
@@ -1,30 +1,160 @@
+            activityId = "activityId",
+            rewindId = "rewindId",
+            downloadId = 100L,
+            siteId = 200L,
+            url = null,
+            published = Date(1609690147756)
+    )
+
     @Before
     fun setUp() {
+        whenever(wizardManager.navigatorLiveData).thenReturn(wizardManagerNavigatorLiveData)
+        whenever(wizardManager.showNextStep()).then {
+            wizardManagerNavigatorLiveData.value = backupDownloadStep
+            Unit
+        }
         viewModel = BackupDownloadViewModel(wizardManager)
     }
 
+    @Test
+    fun `when view model is started, process moves to next step`() {
+        viewModel.start(null)
+
+        verify(wizardManager).showNextStep()
+    }
+
+    @Test
+    fun `when details is finished, process moves to next step`() {
+        viewModel.start(null)
+        // need to clear invocations because nextStep is called on start
+        clearInvocations(wizardManager)
+
+        viewModel.onBackupDownloadDetailsFinished(rewindId, downloadId, published)
+        verify(wizardManager).showNextStep()
+    }
+
+    @Test
+    fun `when progress is finished, process moves to next step`() {
+        viewModel.start(null)
+        clearInvocations(wizardManager)
+
+        viewModel.onBackupDownloadProgressFinished(url)
+        verify(wizardManager).showNextStep()
+    }
+
+    @Test
+    fun `when details is finished, state is updated properly`() {
+        val navigationTargets = initObservers().navigationTargets
+
+        viewModel.start(null)
+        clearInvocations(wizardManager)
+        whenever(wizardManager.showNextStep()).then {
+            wizardManagerNavigatorLiveData.value = BackupDownloadStep.PROGRESS
+            Unit
+        }
+
+        viewModel.onBackupDownloadDetailsFinished(rewindId, downloadId, published)
+
+        assertThat(navigationTargets.last().wizardState.rewindId).isEqualTo(rewindId)
+        assertThat(navigationTargets.last().wizardState.downloadId).isEqualTo(downloadId)
+        assertThat(navigationTargets.last().wizardState.published).isEqualTo(published)
+    }
+
+    @Test
+    fun `when progress is finished, state is updated properly`() {
+        val navigationTargets = initObservers().navigationTargets
+
+        viewModel.start(null)
+        clearInvocations(wizardManager)
+        whenever(wizardManager.showNextStep()).then {
+            wizardManagerNavigatorLiveData.value = BackupDownloadStep.COMPLETE
+            Unit
+        }
+
+        viewModel.onBackupDownloadProgressFinished(url)
+
+        assertThat(navigationTargets.last().wizardState.url).isEqualTo(url)
+    }
+
+    @Test
+    fun `when onBackPressed while in details step, invokes wizard finished with cancel`() {
+        val wizardFinishedObserver = initObservers().wizardFinishedObserver
+
+        viewModel.start(null)
+        clearInvocations(wizardManager)
+
+        whenever(wizardManager.currentStep).thenReturn(0)
```

Comment:
Minor (🔍): Consider using the `BackupDownloadStep.DETAILS.id` instead of using zero directly. `0`, `1` and `2` can be considered magic numbers in this case.

PS: The same applies to the below two `Progress` and `Complete` related tests.